File: 05_capitalize_first_letter.js
// create a function to capitalize first letter

function capitalizeWords(str){
    let words = str.split(" ").map(word =>{
        let firstLetter = word.slice(0,1);
        let restLetter = word.slice(1);
        firstLetter = firstLetter.toUpperCase();
        return `${firstLetter}${restLetter}`;
    });

    return words.join(" ");
}


console.log(capitalizeWords("I got up early today"));
console.log(capitalizeWords("I walked straight to the beach"));
console.log("###############################")

// an effective way
function effectiveCapitalizeWords(str){
    let words = str.split(" ").map(word=>{
        return word.charAt(0).toUpperCase()+ word.slice(1);
    });
    return words.join(" ");
}
console.log(capitalizeWords("I got up early today"));
console.log(capitalizeWords("I walked straight to the beach"));

File: 05_capitalize_first_letter.py
# create a function to capitalize first letter

def capitalizeWords(str):
    result = ' '.join(word[0].upper() + word[1:] for word in str.split())
    return result


print(capitalizeWords("I got up early today"))
print(capitalizeWords("I walked straight to the beach"))
print("#################################")

# an effective way
def effectiveCapitalizeWords(str):
    return str.title()


print(effectiveCapitalizeWords("I got up early today"))
print(effectiveCapitalizeWords("I walked straight to the beach"))